### PR TITLE
Upstream gdb support of analyzing missing symbols

### PR DESCRIPTION
**Pull request by mingweishih**
This PR upstreams #4259 and #4279 on the gdb support of analyzing missing symbols

Signed-off-by: Ming-Wei Shih <mishih@microsoft.com>

### CODE REVIEW

**Comment by mingweishih** [comment]
bors try

**Comment by mingweishih** [comment]
hold the PR and see if we can add tests

File: debugger/gdb-extension/gdb_sgx_plugin.py [hunk 1]
```diff
@@ -381,12 +382,117 @@ def oe_debugger_cleanup():
...
+             name:libname,
+                'imports':imports,
+                'exports':exports
+             }
+        self.libs[libname] = lib
+        self.find_missing_symbols(libname, lib)
+
+missing_symbol_analyzer = None
+
+def oe_debugger_init_hook():
```
Comment (anakrish) [inline debugger/gdb-extension/gdb_sgx_plugin.py:484]: We don't need the debugger hooks and don't need to create the analyzer by default.

**Review by anakrish — APPROVED** [review]
Thanks for upstreaming the debugger work.
This is mainly a tool for Mystikos developers and not end users. There is no need to lock it down with a test. There is also not much value anymore for enabling it by default.

**Review by radhikaj — APPROVED** [review]
bors r+

**Comment by mingweishih** [comment]
> Thanks for upstreaming the debugger work. This is mainly a tool for Mystikos developers and not end users. There is no need to lock it down with a test. There is also not much value anymore for enabling it by default.

Do you think we can make this feature as a plugin can check in to the Mystikos side?
I've done the same with mprotect tracker: https://github.com/deislabs/mystikos/pull/1272/commits/4fab34bd71861877bd3a3e338a6050250495275f

**Comment by mingweishih** [comment]
Will upstream this feature as plugin to the mystikos directly. Closing this PR.

**Pull request closed without merging**